libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -161,6 +161,40 @@ friend class _CUDA_VSTD::__barrier_poll_tester_parity;
         )
     }
 
+    _LIBCUDACXX_NODISCARD_ATTRIBUTE _LIBCUDACXX_INLINE_VISIBILITY
+
+    arrival_token arrive_tx(_CUDA_VSTD::ptrdiff_t arrive_count_update,
+                            _CUDA_VSTD::ptrdiff_t transaction_count_update) {
+#if (_LIBCUDACXX_DEBUG_LEVEL >= 2)
+        _LIBCUDACXX_DEBUG_ASSERT(arrive_count_update >= 0);
+        _LIBCUDACXX_DEBUG_ASSERT(transaction_count_update >= 0);
+        // https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#contents-of-the-mbarrier-object
+        // Maximum value of tx-count is 2^20 -1.
+        _LIBCUDACXX_DEBUG_ASSERT(transaction_count_update <= (1 << 20) - 1);
+        _LIBCUDACXX_DEBUG_ASSERT(__isClusterShared(&__barrier));
+#endif
+        arrival_token __token = {};
+        NV_DISPATCH_TARGET(
+            NV_PROVIDES_SM_90, (
+                if (!__isShared(&__barrier)) {
+                    __trap(); // XXX: We do no support arriving on remote cluster barrier in this method.
```

Comment:
This should be 
```suggestion
                    _LIBCUDACXX_UNREACHABLE(); // XXX: We do no support arriving on remote cluster barrier in this method.
```

Reply:
See I am no barrier expert. In that case we should add an assert but I will defer that to @wmaxey because he has some plans to rework barrier in the near future